includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -3242,10 +3246,7 @@ private function is_platform_payment_method( bool $is_using_saved_payment_method
 		// Make sure the payment method being charged was created in the platform.
 		if (
 			! $is_using_saved_payment_method &&
-			$this->should_use_stripe_platform_on_checkout_page() &&
-			// This flag is useful to differentiate between PRB, blocks and shortcode checkout, since this endpoint is being used for all of them.
-			! empty( $_POST['wcpay-is-platform-payment-method'] ) && // phpcs:ignore WordPress.Security.NonceVerification
-			filter_var( $_POST['wcpay-is-platform-payment-method'], FILTER_VALIDATE_BOOLEAN ) // phpcs:ignore WordPress.Security.NonceVerification,WordPress.Security.ValidatedSanitizedInput.MissingUnslash
+			$this->should_use_stripe_platform_on_checkout_page()
 		) {
```

Comment:
@ricardo we are going to support platform checkout on blocks checkout page also. Do you see any issues or regression if I remove this condition from the check? Asking for your thoughts as you had added this logic initially. :eyes:

Reply:
@Mayisha From the top of my head, I remember having added this check for the reasons described in that comment 🙂. Does the PRB (payment request button) still work from the checkout page after the change?

Also, does the pay for order page still work?

Reply:
@Mayisha props for asking and making sure 🙌.

I tested the PRB here and it didn't work. It throws an error:

> Error: No such PaymentMethod: 'pm_1Lmgk7Qq9iIkB8Vrfac04v6W'

So I guess that check is still important.

To get to the "pay for order page", you need to create a new order from wp-admin, add a product to it, click "Create" and visit the "Customer payment page". This flow works for me.

Reply:
**Steps to reproduce:**
- Enable express (Apple Pay / Google Pay) checkouts from **Payments > Settings**.
- Click to "Customize" and enable it on Checkout, Product page and Cart.
- With a public HTTPS URL (using Jurassic Tube or Ngrok), visit the product, cart or checkout page.
- Try to pay using Google Pay.

Reply:
Thanks @ricardo for testing this and the detailed steps. I will bring this back and add a new condition to support checkout blocks here. :nerd_face: